lib/takeaway.rb

```diff
@@ -0,0 +1,24 @@
+require_relative 'order'
+
+class Takeaway
+
+attr_reader :order, :selected
+
+  def initialize(order = Order.new)
+    @order = order
+  end
+
+  def list_menu
+     order.menu.dishes
+  end
+
+
+  def add_item(dish, qty = 1)
```

Comment:
There's no need to duplicate this functionality.  Construct the order inside an `Order` instance then pass it to the takeaway for completion.